test/raft/randomized_nemesis_test.cc

```diff
@@ -432,16 +451,15 @@ class rpc : public raft::rpc {
 
     template <typename F>
     auto with_gate(F&& f) -> decltype(f()) {
-        try {
-            co_return co_await seastar::with_gate(_gate, std::forward<F>(f));
-        } catch (const gate_closed_exception&) {
-            co_return coroutine::make_exception(raft::stopped_error{});
-        }
+        return seastar::with_gate(_gate, std::forward<F>(f))
+            .handle_exception_type([] (const gate_closed_exception&) -> decltype(f()) {
+                throw raft::stopped_error{};
+            });
```

Comment:
I don't remember right now, but if I had to guess, it's probably because I hit a clang bug which didn't show up in this place earlier for some reason (this is a coroutine template: https://github.com/llvm/llvm-project/issues/49689)

Reply:
ok